src/main/java/org/broadinstitute/hellbender/tools/exome/allelefraction/AFInitializer.java

```diff
@@ -0,0 +1,167 @@
+
+    public AFInitializer(final AFData data) {
+        this.data = data;
+
+        for (int het = 0; het < data.numHets(); het++) {
+                responsibilities.add(new Responsibility(data.count(het)));
+        }
+
+        state = new AFState(BIAS_MEAN_INITIAL, BIAS_RATE_INITIAL, INITIAL_OUTLIER_PROBABILITY, estimateMinorFractions());
+        for (int iteration = 0; iteration < NUMBER_OF_EM_ITERATIONS; iteration++) {
+            performSingleEMIteration();
+        }
+    }
+
+    private double pAltMinor(final int het) { return responsibilities.get(het).get(AFIndicator.ALT_MINOR); }
+    private double pRefMinor(final int het) { return responsibilities.get(het).get(AFIndicator.REF_MINOR); }
+    private double pOutlier(final int het) { return responsibilities.get(het).get(AFIndicator.OUTLIER); }
+    private double pNotOutlier(final int het) { return 1 - pOutlier(het); }
+
+    //perform an E step to update responsibilities using high-level parameters from some AFState
+    private void updateResponsibilities() {
+        for (int segment = 0; segment < data.numSegments(); segment++) {
+            final AFParameters parameters = state.getParameters(segment);
+
+            for (final int het : data.hetsInSegment(segment)) {
+                final AllelicCount count= data.count(het);
+                final double altMinorLogPost = AFModel.hetLogLikelihood(parameters, count, AFIndicator.ALT_MINOR);
+                final double refMinorLogPost = AFModel.hetLogLikelihood(parameters, count, AFIndicator.REF_MINOR);
+                final double outlierLogPost = AFModel.hetLogLikelihood(parameters, count, AFIndicator.OUTLIER);
+                responsibilities.get(het).setFromLogPosteriors(altMinorLogPost, refMinorLogPost, outlierLogPost);
+            }
+        }
+    }
+
+    // M step for minor allele fractions -- set minor fraction to responsibility-weighted total count of
+    // reads in minor allele divided by total reads, ignoring outliers
+    private AFState.MinorFractions estimateMinorFractions() {
+        AFState.MinorFractions result = new AFState.MinorFractions();
+        for (int segment = 0; segment < data.numSegments(); segment++) {
+            final double minorReads = data.hetsInSegment(segment).stream()
+                    .mapToDouble(het -> data.altCount(het) * pAltMinor(het) + data.refCount(het) * pRefMinor(het)).sum();
+            final double totalReads = data.hetsInSegment(segment).stream().mapToDouble(het -> data.readCount(het) * pNotOutlier(het)).sum();
+
+            result.add((minorReads + 1)/(totalReads + 1));
+        }
+        return result;
+    }
+
+    // M step for pOutlier -- set it to the average of all OUTLIER responsibilities
+    private double estimateOutlierProbability() {
+        return responsibilities.stream().mapToDouble(r -> r.get(AFIndicator.OUTLIER)).average().getAsDouble();
+    }
+
+    /**
+     * @param k the order of the desired moment
```

Comment:
Having javadocs for private methods is fine, especially if it could help a future developer.